torchgeo/trainers/regression.py

```diff
@@ -108,13 +130,15 @@ def training_step(self, *args: Any, **kwargs: Any) -> Tensor:
         """
         batch = args[0]
         x = batch["image"]
-        y = batch["label"].view(-1, 1)
+        y = batch[self.target_key]
         y_hat = self(x)
 
-        loss = F.mse_loss(y_hat, y)
+        if y_hat.ndim != y.ndim:
```

Comment:
If `num_outputs=1` the target variable ground truth is missing the necessary channel dim e.g.
- (b,) instead of (b, 1)
- (b, h, w) instead of (b, 1, h, w)

while the output of the models will be:
- (b, 1)
- (b, 1, h, w)